src/orion/benchmark/task/profet/profet_task.py

```diff
@@ -0,0 +1,817 @@
+    size : int
+        Input dimensions for the generated network.
+    idx : int, optional
+        Task idx, by default 0
+
+    Returns
+    -------
+    nn.Module
+        A `nn.Module` with sampled weights.
+    """
+    net = model.get_network(size)
+
+    with torch.no_grad():
+        weights = model.sampled_weights[idx]
+        for parameter, sample in zip(net.parameters(), weights):
+            parameter.copy_(torch.from_numpy(sample))
+    return net
+
+
+def load_task_network(
+    checkpoint_file: Union[str, Path]
+) -> Tuple[nn.Module, np.ndarray]:
+    """Load the result of the `get_task_network` function stored in the pickle file.
+
+    Parameters
+    ----------
+    checkpoint_file : Union[str, Path]
+        Path to a pickle file. The file is expected to contain a serialized dictionary, with keys
+        "benchmark", "size", "network", and "h".
+
+    Returns
+    -------
+    Tuple[nn.Module, np.ndarray]
+        The surrogate model for the objective, as well as an array of sampled task features.
+    """
+    with open(checkpoint_file, "rb") as f:
+        state = pickle.load(f)
+
+    network = get_architecture[state["benchmark"]](state["size"])
+    network.load_state_dict(state["network"])
+    h = state["h"]
+
+    return network, h
+
+
+def compute_identity(size: int = 16, **sample) -> str:
+    """Compute a unique hash out of a dictionary
+
+    Parameters
+    ----------
+    size: int
+        size of the unique hash
+
+    **sample:
+        Dictionary to compute the hash from
+
+    """
+    sample_hash = hashlib.sha256()
+
+    for k, v in sorted(sample.items()):
+        sample_hash.update(k.encode("utf8"))
+
+        if isinstance(v, dict):
+            sample_hash.update(compute_identity(size, **v).encode("utf8"))
+        else:
+            sample_hash.update(str(v).encode("utf8"))
+
+    return sample_hash.hexdigest()[:size]
+
+
+def save_task_network(
+    checkpoint_file: Union[str, Path], benchmark: str, network: nn.Module, h: np.ndarray
+) -> None:
+    checkpoint_file = Path(checkpoint_file)
+    state = dict(
+        benchmark=benchmark,
+        network=network.state_dict(),
+        size=list(network.parameters())[0].size()[1],
+        h=h.tolist(),
+    )
+
+    tmp_file = checkpoint_file.with_suffix(".tmp")
+    with open(tmp_file, "wb") as file:
+        pickle.dump(state, file, protocol=pickle.DEFAULT_PROTOCOL)
+    tmp_file.rename(checkpoint_file)
+
+
+# BUG in pybnn/util/layers.py, where they have self.log_var be a
+# DoubleTensor while x is a FloatTensor! Here we overwrite the
+# forward method, instead of editing the code directly.
+from pybnn.util.layers import AppendLayer
+
+
+@functools.wraps(AppendLayer.forward)
+def _patched_forward(self, x):
+    log_var = self.log_var.type_as(x)
+    return torch.cat((x, log_var * torch.ones_like(x)), dim=1)
+
+
+AppendLayer.forward = _patched_forward
+
+
+InputType = TypeVar("InputType")
+
+
+class ProfetTask(BaseTask, Generic[InputType]):
```

Comment:
Documentation needed. :) Documentation of child class should point to this doc as well so users can better understand what are the different arguments.

Reply:
Like this here inside the child class: https://github.com/Epistimio/orion/blob/develop/src/orion/core/io/database/mongodb.py#L91. It may be simpler if you do it given it is a simple line of code. Maybe you anticipated something more complex?

Reply:
Why is that necessary? Isn't it already explicit that this class extends the base class? In what cases should I add the `seealso` and in what cases is it the documentation tool's job?

Reply:
Right, we should use `autoclass` with option `inherited-members` ([doc](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html)) which will provide the necessary information for methods that are not overwritten by the child class.